Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -444,38 +456,95 @@ extension ActivityStreamPanel {
         })
 
         let deleteFromHistoryAction = ActionOverlayTableViewAction(title: Strings.DeleteFromHistoryContextMenuTitle, iconString: "action_delete", handler: { action in
-            ASOnyxPing.reportDeleteItemEvent(eventInfo)
+            self.telemetry.reportEvent(.Delete, source: pingSource, position: index)
             self.profile.history.removeHistoryForURL(site.url)
         })
 
         let shareAction = ActionOverlayTableViewAction(title: Strings.ShareContextMenuTitle, iconString: "action_share", handler: { action in
             let helper = ShareExtensionHelper(url: siteURL, tab: nil, activities: [])
             let controller = helper.createActivityViewController { completed, activityType in
-                ASOnyxPing.reportShareEvent(eventInfo, shareProvider: activityType)
+                self.telemetry.reportEvent(.Share, source: pingSource, position: index, shareProvider: activityType)
             }
             self.presentViewController(controller, animated: true, completion: nil)
         })
 
         let removeTopSiteAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
-            ASOnyxPing.reportDeleteItemEvent(eventInfo)
+            self.telemetry.reportEvent(.Dismiss, source: pingSource, position: index)
             self.hideURLFromTopSites(site.tileURL)
         })
         
         let dismissHighlightAction = ActionOverlayTableViewAction(title: Strings.RemoveFromASContextMenuTitle, iconString: "action_close", handler: { action in
-            ASOnyxPing.reportDeleteItemEvent(eventInfo)
+            self.telemetry.reportEvent(.Dismiss, source: pingSource, position: index)
             self.hideFromHighlights(site)
         })
 
         var actions = [openInNewTabAction, openInNewPrivateTabAction, bookmarkAction, shareAction]
-        switch eventInfo.source {
-            case .highlights: actions.appendContentsOf([dismissHighlightAction, deleteFromHistoryAction])
-            case .topSites: actions.append(removeTopSiteAction)
-            default: break
+        switch section {
+            case .Highlights: actions.appendContentsOf([dismissHighlightAction, deleteFromHistoryAction])
+            case .TopSites: actions.append(removeTopSiteAction)
+            case .HighlightIntro: break
```

Comment:
We shouldn't rely on the telemetry data type to determine which section to show. I've changed this and separated it at the top of the method.